### PR TITLE
add snappy compression

### PR DESCRIPTION
**Pull request by jangko**

### CODE REVIEW

File: eth_p2p/rlpx.nim [hunk 1]
```diff
@@ -1264,9 +1304,18 @@ proc rlpxAccept*(node: EthereumNode,
     initSecretState(handshake, authMsg, ^ackMsg, result)
 
     let listenPort = transport.localAddress().port
-    await result.hello(devp2pVersion, node.clientId,
-                       node.rlpxCapabilities, listenPort.uint,
-                       node.keys.pubkey.getRaw())
+
+    when defined(useSnappy):
+      let peerProtocolVersion = if result.snappyEnabled: devp2pSnappyVersion.uint
```
Comment (zah) [inline eth_p2p/rlpx.nim]: Let's define a helper template `peer.baseProtocolVersion` that expands to `devp2pVersion` when snappy is disabled and to the branched logic above when snappy is enabled. Then, a lot of the `when` statements in the module would become unnecessary.

File: eth_p2p/rlpx.nim [hunk 2]
```diff
@@ -257,7 +262,16 @@ proc linkSendFailureToReqFuture[S, R](sendFut: Future[S], resFut: Future[R]) =
 proc sendMsg*(peer: Peer, data: Bytes) {.async.} =
   trace "sending msg", peer, msg = getMsgName(peer, rlpFromBytes(data).read(int))
 
-  var cipherText = encryptMsg(data, peer.secretsState)
+  when defined(useSnappy):
+    var cipherText: Bytes
+    if peer.snappyEnabled:
+      let compressed = snappy.compress(data)
```
Comment (zah) [inline eth_p2p/rlpx.nim]: Can't we get away by just using:

```nim
when defined(useSnappy):
  let data = snappy.compress(data)
var cipherText = encryptMsg(data, peer.secretsState)
```

File: eth_p2p/rlpx.nim (hunk 1 above)
Comment (zah) [reply eth_p2p/rlpx.nim]: Also, please align multi-line ifs like this:

```nim
var foo = if x: bar
          else: baz
```

**Comment by jangko** [comment]
now I'm quite confident the logic already meets EIP-706 specifications.

**Pull request merged**